contribute/dev_contribution.rst

```diff
@@ -0,0 +1,110 @@
+
+.. _OpenCMISS-development-process:
+.. figure:: images/OpenCMISSProcesses-DevelopmentProcess.png
+   :align: center
+   :alt: Developer contribution process
+
+   Developer contribution process
+
+GitHub Issue
+============
+
+If an issue does not exist for the required work (e.g. implementation of a feature, fixing of a bug), then create a new one.  The issue is the place to discuss the particulars related to the issue, discussions on determining the scope of the issue or clarification of any points that are unclear.
+
+Labels
+------
+
+A GitHub Issue may be assigned labels by the project administrators to help identify its status at a glance. General labels currently used for OpenCMISS are:
+
+* **Bug**: The issue identifies a malfunction in the current codebase.
+* **Feature**: The issue constitutes a request or plan for a new feature.
+* **Needs tests**: The issue requires test(s) to be complete. This may refer to a bug report, contributed code, comments, etc. in the issue.
+* **Needs documentation**: The issue requires documentation to be complete. This may refer to a bug report, contributed code, comments, etc. in the issue.
+* **Needs reviewing**: The issue requires further review from project participants to be complete. This may refer to a bug report, contributed code, comments, etc. in the issue.
+
+In addition, a **Platform** label may be used to identify the issue as specific to a given platform (Windows/Linux/OS X). **Milestone** labels may be used to project when a feature is expected to be complete and/or indicate the priority of a given issue. Higher priority issues will take precedence and therefore be assigned a more immediate (lower) milestone number.
+
+Topic Branch
+============
+
+A topic or feature branch is a branch that is local to you (and anyone you collaborate with), it is a branch that will not be available from the :term:`prime repository`.  All development work should be carried out on a topic branch, for example any major feature that you work on or minor bug fix. Before creating a local topic branch, pull the latest changes from the :term:`prime repository` develop branch into your develop branch.
+
+.. _OpenCMISS-branching:
+.. figure:: images/OpenCMISSProcesses-GitBranching.png
+   :align: center
+   :alt: Git branching illustration
+
+   Git branching illustration
+
+Following this process will make it easier to have multiple topic branches at once and keep them in-sync with the :term:`prime repository` develop branch.  Which will in turn make it easier to manage multiple pull requests.
+
+The following Git command line commands show an example of how to create a topic branch for fixing a (hypothetical) bug described in `issue #46`::
+
+   git fetch prime develop
+   git checkout develop # Not required if already on develop branch
+   git merge prime/develop
+   git checkout -b issue46
```

Comment:
I would suggest we recommend users name their branches starting with the issue number followed by a word or words to succinctly describe it; 'issue' is redundant. Some examples I've used: 56_refactor_fe, 36_derivatives.